offchain-modules/packages/x/src/handlers/ckb.ts

```diff
@@ -743,9 +858,151 @@ export class CkbHandler {
+        senderLockscript.hashType === committeeMultisigLockscript.hash_type &&
+        senderLockscript.args === committeeMultisigLockscript.args)
+    ) {
+      logger.error(`sender should not be committee multisig cell`);
+      return null;
+    }
+
+    if (tx.inputs.length > 1) {
+      for (let i = 1; i < tx.inputs.length; i++) {
+        const thisPreviousOutput = nonNullable(tx.inputs[i].previousOutput);
+        const thisPreHash = thisPreviousOutput.txHash;
+        const thisTxPrevious = await this.ckb.rpc.getTransaction(thisPreHash);
+        if (thisTxPrevious == null) {
+          return null;
+        }
+        const thisSenderLockscript = thisTxPrevious.transaction.outputs[Number(thisPreviousOutput.index)].lock;
+        if (
+          !thisSenderLockscript ||
+          thisSenderLockscript.codeHash !== senderLockscript.codeHash ||
+          thisSenderLockscript.hashType !== senderLockscript.hashType ||
+          thisSenderLockscript.args !== senderLockscript.args
+        ) {
+          logger.error(`inputs contain different lockscripts`);
+          return null;
+        }
+      }
+    }
+
+    if (
+      recipientLockscript.codeHash !== committeeMultisigLockscript.code_hash ||
+      recipientLockscript.hashType !== committeeMultisigLockscript.hash_type ||
+      recipientLockscript.args !== committeeMultisigLockscript.args
+    ) {
+      logger.error(`the first output must be committee multisig`);
+      return null;
+    }
+
+    if (tx.outputs.length > 1) {
+      for (let i = 1; i < tx.outputs.length; i++) {
+        const thisLockscript = nonNullable(tx.outputs[i].lock);
+        if (
+          thisLockscript.codeHash === committeeMultisigLockscript.code_hash &&
+          thisLockscript.hashType === committeeMultisigLockscript.hash_type &&
+          thisLockscript.args === committeeMultisigLockscript.args
+        ) {
+          logger.error(`the outputs except first must not be committee multisig`);
+          return null;
+        }
+      }
+    }
+
+    let amount = BigInt(0);
+    let bridgeFee = BigInt(ForceBridgeCore.config.eth.lockNervosAssetFee);
+    let assetIdent = '0x0000000000000000000000000000000000000000000000000000000000000000';
+    if (
+      recipientTypescript &&
+      recipientTypescript.codeHash === sudtTypescript.codeHash &&
+      recipientTypescript.hashType === sudtTypescript.hashType
+    ) {
+      amount = utils.readBigUInt128LE(tx.outputsData[0]);
+      assetIdent = utils.computeScriptHash({
+        code_hash: recipientTypescript.codeHash,
+        hash_type: recipientTypescript.hashType,
+        args: recipientTypescript.args,
+      });
+      const nervosAsset = ForceBridgeCore.config.eth.nervosAssetWhiteList.find(
+        (asset) => asset.typescriptHash === assetIdent,
+      );
+      if (!nervosAsset) {
+        logger.error(`nervos asset ${nervosAsset} not in nervos asset white list`);
+        return null;
+      }
```

Comment:
If asset user locked not in white list, we will record it in db, rather than ignore it. And when collector create ethereumMint record, we will check if the asset is in white list, ref: https://github.com/nervosnetwork/force-bridge/blob/a85d9b8e4ea4e4d55b55f2f86e22380e496e1f04/offchain-modules/packages/x/src/handlers/eth.ts#L224

Reply:
added checkLock after createCkbLock in db and moved white list check into checkLock